types/any/test/types_test.go

```diff
@@ -0,0 +1,204 @@
+package test
+
+import (
+	types "github.com/cosmos/gogoproto/types/any"
+	"github.com/cosmos/gogoproto/types/any/internal"
+	"strings"
+	"testing"
+
+	"github.com/cosmos/gogoproto/jsonpb"
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/stretchr/testify/require"
+)
+
+func NewTestInterfaceRegistry() internal.InterfaceRegistry {
+	registry := internal.NewInterfaceRegistry()
+	RegisterInterfaces(registry)
+	return registry
+}
+
+func RegisterInterfaces(registry internal.InterfaceRegistry) {
+	registry.RegisterInterface("Animal", (*Animal)(nil))
+	registry.RegisterImplementations(
+		(*Animal)(nil),
+		&Dog{},
+		&Cat{},
+	)
+	registry.RegisterImplementations(
+		(*HasAnimalI)(nil),
+		&HasAnimal{},
+	)
+	registry.RegisterImplementations(
+		(*HasHasAnimalI)(nil),
+		&HasHasAnimal{},
+	)
+}
+
+func TestAnyPackUnpack(t *testing.T) {
+	registry := internal.NewInterfaceRegistry()
+
+	spot := &Dog{Name: "Spot"}
+	var animal Animal
+
+	// with cache
+	any, err := types.NewAnyWithCacheWithValue(spot)
+	require.NoError(t, err)
+	require.Equal(t, spot, any.GetCachedValue())
+	err = registry.UnpackAny(any, &animal)
```

Comment:
It doesn't make sense to me that we should duplicate code from the SDK (InterfaceRegistry) and then exercise here in test.  This test package should just test the functions of `any.pb.go` not consumers in other packages, especially when it means copy-pasting code, right?